### PR TITLE
Add validation to prevent buffer overflow in Qwen2VL grid dimensions

### PR DESCRIPTION
**Pull request by apsonawane**
This pull request introduces important validation checks for grid tensor dimensions in the Qwen2VL position input logic to prevent out-of-bounds memory access and potential overflows. It also adds a new test file documenting the expected validation behavior for various edge cases, improving the robustness and safety of the code.

**Validation and bounds checking improvements:**

* Added validation in `SetGridTensors` to ensure `image_grid_thw` and `video_grid_thw` tensors have element counts divisible by 3, contain non-negative values, and do not exceed a maximum allowed dimension (`kMaxGridDim = 16384`). This prevents invalid or malicious input from causing overflows or undefined behavior.
* Added a runtime check in `CreateAndInitialize3DPositionIDs` to ensure that the number of vision tokens (`vision_len`) does not exceed the available sequence length, throwing an exception if this would lead to out-of-bounds writes.

**Testing and documentation:**

* Introduced a new test file `test/qwen2vl_position_inputs_test.cpp` with test cases documenting the expected rejection of negative, excessively large, or incorrectly sized grid dimensions, as well as acceptance of valid input. These tests serve as documentation and placeholders for future integration tests.

### CODE REVIEW

File: src/models/position_inputs.cpp [hunk 1]
```diff
@@ -648,6 +648,36 @@ Qwen2VLPositionInputs::Qwen2VLPositionInputs(const Model& model, State& state, D
 void Qwen2VLPositionInputs::SetGridTensors(const std::shared_ptr<Tensor>& image_grid_thw,
                                            const std::shared_ptr<Tensor>& video_grid_thw,
                                            const std::shared_ptr<Tensor>& second_per_grid_ts) {
+  // Validate grid tensor dimensions to prevent overflow during position computation
+  constexpr int64_t kMaxGridDim = 16384;  // Conservative upper bound to prevent overflow
```
Comment (kunal-vaishnavi) [inline src/models/position_inputs.cpp]: How likely is it that this upper bound will be exceeded?

File: test/qwen2vl_position_inputs_test.cpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <gtest/gtest.h>
+// Test that SetGridTensors rejects negative grid dimensions
+TEST(Qwen2VLPositionInputsTest, RejectsNegativeGridDimensions) {
+  GTEST_SKIP() << "TODO: wire this to a real Qwen2VLPositionInputs validation path.";
```
Comment (kunal-vaishnavi) [inline test/qwen2vl_position_inputs_test.cpp]: Can you implement these TODOs in this PR?

File: src/models/position_inputs.cpp (hunk 1 above)
Comment (apsonawane) [reply src/models/position_inputs.cpp]: It is very rare the actual grid values are much smaller.

**Review by kunal-vaishnavi — APPROVED** [review]
